src/prognostic_equations/vertical_diffusion_boundary_layer.jl

```diff
@@ -77,23 +129,87 @@ function vertical_diffusion_boundary_layer_tendency!(
 )
     ᶜρ = Y.c.ρ
     FT = Spaces.undertype(axes(ᶜρ))
-    (; ᶜp, ᶜspecific, sfc_conditions) = p.precomputed # assume ᶜts and ᶜp have been updated
-    (; C_E) = p.atmos.vert_diff
+    (; params) = p
+    thermo_params = CAP.thermodynamics_params(params)
+    (; ᶜts, ᶜp, ᶜspecific, sfc_conditions) = p.precomputed # assume ᶜts and ᶜp have been updated
+
+    κ = CAP.von_karman_const(params)
+    grav = CAP.grav(params)
+    
+    # TODO : Move to ClimaParameters
+    z₀ = FT(3.21e-5)
+    f_b = FT(0.1)
+    Ri_c = FT(1)
 
     ᶠgradᵥ = Operators.GradientC2F() # apply BCs to ᶜdivᵥ, which wraps ᶠgradᵥ
 
+    ᶠρK_E = p.scratch.ᶠtemp_scalar
+    θ_v = p.scratch.ᶜtemp_scalar
+    Ri = p.scratch.ᶜtemp_scalar_2
+    dz_local = p.scratch.ᶜtemp_scalar_3
+    θ_v_sfc = p.scratch.ᶠtemp_field_level
+    Ri_a = p.scratch.temp_field_level_2
+
+    # Compute boundary layer height
     FT = eltype(Y)
+    ᶠts_sfc = sfc_conditions.ts
+    ᶜz = Fields.coordinate_field(Y.c).z
+    zₐ = Fields.level(ᶜz, 1)
     interior_uₕ = Fields.level(Y.c.uₕ, 1)
-    ᶠp = ᶠρK_E = p.scratch.ᶠtemp_scalar
-    @. ᶠp[colidx] = ᶠinterp(ᶜp[colidx])
     ᶜΔz_surface = Fields.Δz_field(interior_uₕ)
-    @. ᶠρK_E[colidx] =
-        ᶠinterp(Y.c.ρ[colidx]) * eddy_diffusivity_coefficient(
-            C_E,
+    @. θ_v[colidx] = TD.virtual_pottemp(thermo_params, ᶜts[colidx])
+    θ_a = Fields.level(θ_v[colidx], 1)
```

Comment:
```suggestion
    θ_v_a = Fields.level(θ_v[colidx], 1)
```
(I think this is more clear)